core/src/com/gaalf/game/precreatedEntities/balls/SquareBall.java

```diff
@@ -14,7 +15,9 @@
     SquareBall(String playerName, int playerNumber, TiledMap tiledMap, World world, GameAssetManager assetManager) {
         super(playerName, playerNumber, assetManager);
         TransformComponent transformComponent = addTransformComponent(tiledMap, 1.5f);
-        TextureComponent textureComponent = addTextureComponent((Texture)assetManager.manager.get(assetManager.golfBallSprite));
+        TextureAtlas textureAtlas = assetManager.manager.get(assetManager.ballSpriteAtlas);
+        Sprite sprite = textureAtlas.createSprite("Square");
+        TextureComponent textureComponent = addTextureComponent(sprite);
```

Comment:
Maybe move the creation of the sprite inside the addTextureComponent() function, and just take the string as a parameter to the function instead?